gdal/alg/contour.cpp

```diff
@@ -615,11 +615,12 @@ CPLErr GDALContourGenerateEx( GDALRasterBandH hBand, void *hLayer,
     {
         if ( polygonize )
         {
-            PolygonContourWriter w( &oCWI );
+            int bSuccess; 
+            PolygonContourWriter w( &oCWI, GDALGetRasterMinimum( hBand, &bSuccess ) );
             typedef PolygonRingAppender<PolygonContourWriter> RingAppender;
             RingAppender appender( w );
             if ( ! fixedLevels.empty() ) {
-                FixedLevelRangeIterator levels( &fixedLevels[0], fixedLevels.size() );
+                FixedLevelRangeIterator levels( &fixedLevels[0], fixedLevels.size(), GDALGetRasterMaximum( hBand, &bSuccess ) );
```

Comment:
just a word of warning: GDALGetRasterMaximum() will return the true maximum value only if statistics have been computed. Otherwise it will return the maximum value of the data type (or 4294967295.0 for floating point data types)
Similarly for GDALGetRasterMinimum().

Reply:
Maybe it would be better to use GetStatistics to force computation of real min/max? But I'm concern about performance issue.

Reply:
> Maybe it would be better to use GetStatistics to force computation of real min/max? But I'm concern about performance issue.

I guess that when contouring the algorithm explores all cell values, so it could get them that way. But I don't think there's a strong need for that. GetRasterMinimum()/Maximum() is probably good enough for now. Just wanting to draw attention on this